cli/src/main/java/keywhiz/cli/commands/UpdateAction.java

```diff
@@ -58,6 +58,9 @@ public UpdateAction(UpdateActionConfig config, KeywhizClient client,
     byte[] content = {};
     if (config.contentProvided) {
       content = readSecretContent();
+      if (content.length == 0) {
+        throw new IllegalArgumentException("Secret content must be provided if --content flag is present; check inputs");
+      }
```

Comment:
Before this change, `partialUpdateSecret` would tell the server that content had _not_ been provided if the content was empty, so the contents would silently not be reset (but they would not be cleared, even if the `--content` flag had been set).
```
keywhizClient.updateSecret(secretName, config.description != null,
          config.getDescription(), content.length > 0, content, config.json != null,
          config.getMetadata(mapper), config.expiry != null, config.getExpiry());
```